src/integration-test/java/com/commercetools/project/sync/CliRunnerIT.java

```diff
@@ -303,6 +306,131 @@ void run_WithSyncAsArgumentWithAllArg_ShouldExecuteAllSyncers() {
+
+        assertCustomerSyncerLoggingEvents(syncerTestLogger, 1);
+
+        assertCustomersAreSyncedCorrectly(postTargetClient);
+
+        final String sourceProjectKey = postSourceClient.getConfig().getProjectKey();
+
+        assertLastSyncCustomObjectExists(
+            postTargetClient, sourceProjectKey, "customerSync", "runnerName");
+
+        lastModifiedTime =
+            getCustomObjectLastModifiedTime(postTargetClient, "customerSync", "runnerName");
+        updateCustomerSourceObject(postSourceClient);
+      }
+    }
+
+    try (final SphereClient targetClient = createClient(CTP_TARGET_CLIENT_CONFIG)) {
+      try (final SphereClient sourceClient = createClient(CTP_SOURCE_CLIENT_CONFIG)) {
+
+        final SyncerFactory syncerFactory =
+            SyncerFactory.of(() -> sourceClient, () -> targetClient, Clock.systemDefaultZone());
+
+        // test
+        CliRunner.of().run(new String[] {"-s", "customers"}, syncerFactory);
+      }
+    }
+
+    // create clients again (for assertions and cleanup), since the run method closes the clients
+    // after execution is done.
+    try (final SphereClient postSourceClient = createClient(CTP_SOURCE_CLIENT_CONFIG)) {
+      try (final SphereClient postTargetClient = createClient(CTP_TARGET_CLIENT_CONFIG)) {
+        assertUpdatedCustomersAreSyncedCorrectly(postSourceClient, postTargetClient);
+
+        assertUpdatedCustomObjectTimestampAfterSync(
+            postTargetClient, "customerSync", "runnerName", lastModifiedTime);
+      }
+    }
+  }
+
+  private void assertUpdatedCustomObjectTimestampAfterSync(
+      @Nonnull final SphereClient targetClient,
+      @Nonnull final String syncModuleName,
+      @Nonnull final String runnerName,
+      ZonedDateTime lastModifiedTime) {
+    final PagedQueryResult<CustomObject<LastSyncCustomObject>> lastSyncResult =
+        getCustomObjectPagedQueryResult(targetClient, syncModuleName, runnerName);
+
+    assertThat(lastModifiedTime)
+        .isBefore(lastSyncResult.getResults().get(0).getValue().getLastSyncTimestamp());
+  }
+
+  private ZonedDateTime getCustomObjectLastModifiedTime(
+      @Nonnull final SphereClient targetClient,
+      @Nonnull final String syncModuleName,
+      @Nonnull final String runnerName) {
+    final PagedQueryResult<CustomObject<LastSyncCustomObject>> lastSyncResult =
+        getCustomObjectPagedQueryResult(targetClient, syncModuleName, runnerName);
+
+    return lastSyncResult.getResults().get(0).getValue().getLastSyncTimestamp();
+  }
+
+  static void updateCustomerSourceObject(@Nonnull final SphereClient sourceProjectClient) {
+
+    final PagedQueryResult<Customer> customerPagedQueryResult =
+        sourceProjectClient
+            .execute(
+                CustomerQuery.of()
+                    .withPredicates(QueryPredicate.of(format("key=\"%s\"", RESOURCE_KEY))))
+            .toCompletableFuture()
+            .join();
+    Customer customer = customerPagedQueryResult.getResults().get(0);
```

Comment:
```suggestion
    final Customer customer = customerPagedQueryResult.getResults().get(0);
```

General rule - either use `final` everywhere or don't use it at all :)